### ISSUE TITLE
Respect AZERTY keybord layout in config file

### ISSUE THREAD

**Issue by theottm** [xremap/xremap#552]
I have a French keyboard (AZERTY). When installing Ubuntu I can choose the keyboard layout accordingly so that the letters I see on my keyboard correspond to what is interpreted by the computer.

It seems that xremap is  ignoring this layout. So when I type `A` on my keyboard I get a `Q` when checking at the output of `sudo RUST_LOG=debug xremap ~/.config/xremap/emacs.yml`, which is the key I would get in the english layout.

If I want my xremap config to work, I have to write a translated version of it. For example instead of `C-a : home`, I have to write `C-q : home`, since what I see as A on my keyboard is seen as Q by xremap. This would make my config unmanageable.

Is there a solution on xremap's side ? Can I maybe chain remaps  and have a master remap that acts as the keyboard layout? Is it possible to make xremap get its input after the keyboard layout has been applied ?

**Comment by wistoft** [comment]
I have the same problem. I've learned to live with it, because I don't believe it's possible, without changes to xremap.

You can't really chain remaps for this. Modmap is applied before keymap, but that's only for input. And it would mess up, because both xremap and the OS would apply the keyboard layout.

Xremap doesn't know anything of the keyboard layout, as it is now. But a keyboard map could be applied before remapping, and then inversely applied before emitting. With a custom keyboard layout in the configuration file, the problem om fetching it from the OS is gone, and it would be an easier task to do.

**Comment by theottm** [comment]
Hi, thanks for your answer.

So your suggest that
> a keyboard map could be applied before remapping, and then inversely applied before emitting

Could you elaborate on this? How would I apply the keyboard map ? With modmap? With another instance of xremap? Something else? If I understand you correctly you want a mapping $$f$$ from the native mapping to my custom keybord layout and the apply something like  $$f xf^{-1}$$ where $$x$$ is xremap?

**Comment by wistoft** [comment]
Yes that's the idea. I thought it best to do it in xremap, because it's actually quite simple. And I've created a [PR](https://github.com/xremap/xremap/pull/553) you test.

Doing it with instances of xremap would probably work, but it's too cumbersome, I think.
